textBasedAdventure.js

```diff
@@ -1,5 +1,321 @@
+  }
+}
+
+function inTownAfterSwimming() {
+  console.log("\n");
+    console.log("You swam across the lake and reached a town. ");
+  console.log(
+    "You feel hungry and want to get a bite. You have few different choices."
+  );
+  let res = readline.keyIn(
+    "Would you eat at McDonalds (m), Chinese food bar (c) or Seafood restaurant (s)? ",
+    { limit: 'mcs' }
+  ).toLowerCase();
+  if (res === "m") {
+    inMcDonalds();
+  } else if (res === "c") {
+    inChineseRest();
+  } else if (res === "s") {
+    inSeafoodRest();
+  }
+}
+
+function inTownAfterBoat() {
+ console.log("\n");
+    console.log("You crossed the lake by boat and reached a town. ");
+  console.log(
+    "You feel hungry and want to get a bite. You have few different choices."
+  );
+  let res = readline.keyIn(
+    "Would you eat at McDonalds (m), Chinese food bar (c) or Seafood restaurant (s)? ",
+    { limit: 'mcs' }
+  ).toLowerCase();
+  if (res === "m") {
+    inMcDonalds();
+  } else if (res === "c") {
+    inChineseRest();
+  } else if (res === "s") {
+    inSeafoodRest();
+  }
+}
+
+function inTownAfterHelicopter() {
+  console.log("\n");
+    console.log("You crossed the lake by helicopter and reached a town. ");
+  console.log(
+    "You feel hungry and want to get a bite. You have few different choices."
+  );
+  let res = readline.keyIn(
+    "Would you eat at McDonalds (m), Chinese food bar (c) or Seafood restaurant (s)? ",
+    { limit: 'mcs' }
+  ).toLowerCase();
+  if (res === "m") {
+    inMcDonalds();
+  } else if (res === "c") {
+    inChineseRest();
+  } else if (res === "s") {
+    inSeafoodRest();
+  }
+}
+
+function inMcDonalds() {
+  console.log("\n");
+    console.log("You've got your food and now need to choose the seating.");
+  let res = readline.keyIn(
+    "Would you like a seat in the center of dining room on the first floor (f), by the window on the second floor (s) or at the outdoor seating place (o)? ",
+    { limit: 'fso'}
+  ).toLowerCase();
+  if (res === "f") {
+    console.log("Enjoy your food!");
+    outsideRestaurant();
+  } else if (res === "s") {
+    console.log("Enjoy your food!");
+    outsideRestaurant();
+  } else if (res === "o") {
+    console.log("Enjoy your food!");
+    outsideRestaurant();
+  }
+}
+
+function inChineseRest() {
+    console.log("\n");
+    console.log("You've got your food and now need to choose the seating.");
+    let res = readline.keyIn(
+      "Would you like a seat in the center of dining room on the first floor (f), by the window on the second floor (s) or at the outdoor seating place (o)? ",
+      { limit: 'fso' }
+    ).toLowerCase();
+    if (res === "f") {
+      console.log("Enjoy your food!");
+      outsideRestaurant();
+    } else if (res === "s") {
+      console.log("Enjoy your food!");
+      outsideRestaurant();
+    } else if (res === "o") {
+      console.log("Enjoy your food!");
+      outsideRestaurant();
+    }
+  }
+
+  function inSeafoodRest() {
```

Comment:
These restaurant functions all do the same thing. I think you're probably doing this just to demonstrate that you can go different paths as in different functions. In reality (code wise), if they all do the same thing, they should just be one function)